cgmanifests/submodules/cgmanifest.json

```diff
@@ -0,0 +1,655 @@
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "38f0cfc28d984cba8b368695491f9cc45bca2363",
+          "repositoryUrl": "https://github.com/boostorg/lockfree.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/boost/lockfree"
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "ac67f5952e6c849e461d7651ffb4fa0e7f380bb0",
+          "repositoryUrl": "https://github.com/boostorg/mpl.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/boost/mpl"
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "5d3e39325e13a74645b60af33b16e8369f5ee61d",
+          "repositoryUrl": "https://github.com/boostorg/parameter.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/boost/parameter"
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "aa94d69718a8d4d0a6fd80cb1e62ed62d9b23d83",
+          "repositoryUrl": "https://github.com/boostorg/predef.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/boost/predef"
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "d135cad759062ba646cb4bc40c163e590d381001",
+          "repositoryUrl": "https://github.com/boostorg/preprocessor.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/boost/preprocessor"
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "feaaf3d14ab42d1f248d2b627a60c446fa9f2600",
+          "repositoryUrl": "https://github.com/boostorg/static_assert.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/boost/static_assert"
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "efc694d679263652520d965b94debbec69a17d85",
+          "repositoryUrl": "https://github.com/boostorg/type_traits.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/boost/type_traits"
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "05dda09fd3b4312594144b5a8617ba6fc694a2d6",
+          "repositoryUrl": "https://github.com/boostorg/utility.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/boost/utility"
+      }
+    },
+    {
+      "component": {
+        "type": "git",
+        "git": {
+          "commitHash": "554d8f538e4b0004152c75a9eb72d99f8c0f914c",
+          "repositoryUrl": "https://github.com/eigenteam/eigen-git-mirror.git"
+        },
+        "comments": "git submodule at cmake/external/horovod/third_party/eigen"
```

Comment:
Is it possible to remove some unused components from here?
Like, the eigen from github. We're not using it.

Reply:
It is possible. However, we are still cloning whatever submodules are here. If it doesn't hurt listing them here I'd rather leave them. If we ignore some, we need to keep track of which ones we are ignoring and make sure to not inadvertently use them in the future.

---
In reply to: [425492748](https://github.com/microsoft/onnxruntime/pull/3933#discussion_r425492748) [](ancestors = 425492748)

Reply:
Regarding this eigen in particular, I believe there are plans to remove the horovod dependency. When that happens, this should go away too.

---
In reply to: [425492748](https://github.com/microsoft/onnxruntime/pull/3933#discussion_r425492748) [](ancestors = 425492748)

Reply:
"we need to keep track of which ones we are ignoring"

We may need to do it in the future. Unused submodules can generate unnecessary security warnings.

Reply:
sounds good. it should be straightforward to check a list of paths to ignore in the generate script.